### PR TITLE
Unify decomp registries into global_decomposition_table

### PR DESCRIPTION
**Pull request by SherlockNoMad**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #86857

### CODE REVIEW

**Comment by SherlockNoMad** [comment]
> Will pre_autograd or post_autograd table be used with `config.use_functionalize = False`?

Yes.

**Comment by IvanYashchuk** [comment]
> > Will pre_autograd or post_autograd table be used with `config.use_functionalize = False`?
> 
> Yes.

Both of them or one of them?

**Comment by SherlockNoMad** [comment]
> > > Will pre_autograd or post_autograd table be used with `config.use_functionalize = False`?
> > 
> > 
> > Yes.
> 
> Both of them or one of them?

Both of them.

File: torch/_decomp/__init__.py [hunk 1]
```diff
@@ -23,7 +33,9 @@
 _disabled_meta_decomps = set()
 
 
-def register_decomposition(aten_op, registry=None, *, disable_meta: bool = False):
+def register_decomposition(
+    aten_op, registry=None, *, type="post_autograd", disable_meta: bool = False
```
Comment (ezyang) [inline torch/_decomp/__init__.py:37]: You were planning to get rid of `disable_meta` too, is that right?

**Review by ezyang — APPROVED** [review]
sure

**Pull request closed without merging**